src/sparseml/pytorch/utils/mfac_helpers.py

```diff
@@ -275,54 +280,19 @@ def __init__(self, grads, block_size, damp=1e-5, devices=None):
         self._devices = devices or ["cpu"]
 
         self._fisher_inv_blocks = []
-        lock = threading.Lock()
-
-        # run block computations in parallel across devices
-        threads = [None] * len(self._devices)
-        thread_fisher_inv_blocks = [None] * len(self._devices)
 
-        def _compute_fisher_inv_block(block_start_idx, thread_idx_):
+        _LOGGER.debug("Starting FisherInverseFastBlock")
+        for block_start_idx in range(0, grads.shape[1], self._block_size):
```

Comment:
is there any way to take advantage of multiple gpus here without using an external api?

Reply:
In my research I haven't been able to find another way. It seems in general there are two options - either code your own interactions with the CUDA kernel or rely on ones created by others (e.g. DPP in PyTorch). But I'd be curious to hear if anyone else on the team has had experience with this.